client.go

```diff
@@ -0,0 +1,796 @@
+			reject(err)
+		}
+
+		resolve(members)
+	}()
+
+	return promise
+}
+
+// Srandmember returns a random element from the set value stored at key.
+func (c *Client) Srandmember(key string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		element, err := c.redisClient.SRandMember(c.vu.Context(), key).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(element)
+	}()
+
+	return promise
+}
+
+// Spop removes and returns a random element from the set value stored at key.
+func (c *Client) Spop(key string) *goja.Promise {
+	promise, resolve, reject := c.makeHandledPromise()
+
+	go func() {
+		element, err := c.redisClient.SPop(c.vu.Context(), key).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(element)
+	}()
+
+	return promise
+}
+
+// SendCommand sends a command to the redis server.
+func (c *Client) SendCommand(command string, args ...interface{}) *goja.Promise {
+	var doArgs []interface{}
+	doArgs = append(doArgs, command)
+	doArgs = append(doArgs, args...)
+
+	promise, resolve, reject := c.makeHandledPromise()
+
+	if err := c.isSupportedType(1, args...); err != nil {
+		reject(err)
+		return promise
+	}
+
+	go func() {
+		cmd, err := c.redisClient.Do(c.vu.Context(), doArgs...).Result()
+		if err != nil {
+			reject(err)
+		}
+
+		resolve(cmd)
+	}()
+
+	return promise
+}
+
+// makeHandledPromise will create a promise and return its resolve and reject methods,
+// wrapped in such a way that it will block the eventloop from exiting before they are
+// called even if the promise isn't resolved by the time the current script ends executing.
+func (c *Client) makeHandledPromise() (*goja.Promise, func(interface{}), func(interface{})) {
+	runtime := c.vu.Runtime()
+	callback := c.vu.RegisterCallback()
+	p, resolve, reject := runtime.NewPromise()
+
+	return p, func(i interface{}) {
+			// more stuff
+			callback(func() error {
+				resolve(i)
+				return nil
+			})
+		}, func(i interface{}) {
+			// more stuff
+			callback(func() error {
+				reject(i)
+				return nil
+			})
+		}
+}
+
+// isSupportedType returns whether the provided arguments are of a type
+// supported by the redis client.
+//
+// isSupportedType should report type errors with arguments in the correct
+// position. To be able to accurately report the argument position in the larger
+// context of a call to a redis function, the `offset` argument allows to indicate
+// the amount of arguments present in front of the ones we provide to `isSupportedType`.
+// For instance, when calling `set`, which takes a key, and a value argument,
+// isSupportedType applied to the value should eventually report an error with
+// the argument in position 1.
+func (c *Client) isSupportedType(offset int, args ...interface{}) error {
+	for idx, arg := range args {
+		switch arg.(type) {
+		case string, int, int64, float64, bool:
+			return nil
+		default:
+			return fmt.Errorf("unsupported type: %T for argument in position: %d", arg, idx+offset)
```

Comment:
In the way this is currently used, this is 0-indexed which while popular among programmers, might not be what people expect from this. 

I guess we can change `position` to `index` and that likely will be enough, or we can note that this is 0-indexed somewhere :thinking: 

Making it not 0-indexed has the problem reversed - some people will expect it is 0-indexed :roll_eyes: so we will still need to explain it somehow. Preferably contained within the message

Reply:
I came to the same conclusion 🤝 Renaming to `index` and documenting it is a good idea indeed 💡